workers/work.go

```diff
@@ -0,0 +1,70 @@
+package workers
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/kirikami/go_db_extract/config"
+	"github.com/kirikami/go_db_extract/database"
+	"github.com/kirikami/go_db_extract/services"
+)
+
+type Work struct {
+	Function  func()
+	Completed bool
+}
+
+func NewWork(function func()) *Work {
+
+	return &Work{function, false}
+
+}
+
+func Worker(in chan *Work, out chan *Work) {
+
+	for {
+		work := <-in
+		work.Function()
+		work.Completed = true
+		out <- work
+	}
+
+}
+
+func DbWork(configs *config.TomlConfig) {
+	pending := make(chan *Work)
+	done := make(chan *Work)
+	errors := make(chan error)
+	result := make(chan services.Result, 1)
+
+	databases := len(configs.Database)
+
+	for _, config := range configs.Database {
+		c := config
+		db := database.MustNewDatabase(c)
+		dumpDatabase := func() {
+			services.ArchiveDatabase(db, c, result, errors)
+		}
```

Comment:
Instead of creating special variable by dangerous construction `c := config`(dangerous because people usually removing it on first refactoring), you can use closure with arguments:
```
func(localConfig Config) {
   // here variable localConfig will not change on next loop iteration, because in next iteration will create another closure
   db := database.MustNewDatabase(localConfig)
}(c)
```